controllers/gitrepository_controller.go

```diff
@@ -113,8 +112,9 @@ type GitRepositoryReconciler struct {
 	kuberecorder.EventRecorder
 	helper.Metrics
 
-	Storage        *Storage
-	ControllerName string
+	Storage                     *Storage
+	ControllerName              string
+	ManagedTransportsRegistered bool
```

Comment:
Even though transport registration at `git2go`, I think we can be confident about it since, if the transports fail to register, a non-nil error would be returned. I think it's safe to assume that if the error is nil, our transports have been registered. `UseManagedTransport` implies the same thing as `GitManagedTransport=true`, which could cause confusion, in terms of code readability.

Reply:
Also, based on this discussion, it may be better to have a description of this field, what it means. Unlike other public fields, the intention of this field may not be clear.